apps/paradym/app/_layout.tsx

```diff
@@ -29,20 +27,19 @@ export const unstable_settings = {
 }
 
 export default function HomeLayout() {
-  const [agent, setAgent] = useState<FullAppAgent>()
-  const [isMediationConfigured, setIsMediationConfigured] = useState(false)
+  const [agent, setAgent] = useState<ParadymAppAgent>()
   const hasInternetConnection = useHasInternetConnection()
 
   const [agentInitializationFailed, setAgentInitializationFailed] = useState(false)
   const toast = useToastController()
   const { top } = useSafeAreaInsets()
-  const [isSettingUpMediation, setIsSettingUpMediation] = useState(false)
   useTransparentNavigationBar()
 
-  // Enable message pickup when mediation is configured and internet connection is available
-  useMessagePickup({
-    isEnabled: hasInternetConnection && isMediationConfigured,
-    agent,
+  // Only setup mediation if the agent is a paradym agent
+  useMediatorSetup({
```

Comment:
This is totally ok for now, no need to spend more time on it, but this could also use the feature flags right? or is there a specific reason you didn't use it?

Reply:
No not really. I can add it to the check as well.